sync/handlers/signature_request_test.go

```diff
@@ -0,0 +1,98 @@
+	"github.com/ava-labs/subnet-evm/plugin/evm/warp"
+	"github.com/ava-labs/subnet-evm/sync/handlers/stats"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSignatureHandler(t *testing.T) {
+	database := memdb.New()
+	snowCtx := snow.DefaultContextTest()
+	blsSecretKey, err := bls.NewSecretKey()
+	require.NoError(t, err)
+
+	snowCtx.TeleporterSigner = teleporter.NewSigner(blsSecretKey, snowCtx.ChainID)
+	warpBackend := warp.NewWarpBackend(snowCtx, database, 100)
+
+	msg, err := teleporter.NewUnsignedMessage(snowCtx.ChainID, snowCtx.CChainID, []byte("test"))
+	require.NoError(t, err)
+
+	messageID := hashing.ComputeHash256Array(msg.Bytes())
+	require.NoError(t, warpBackend.AddMessage(context.Background(), msg))
+	signature, err := warpBackend.GetSignature(context.Background(), messageID)
+	require.NoError(t, err)
+	unknownMessageID := ids.GenerateTestID()
+
+	mockHandlerStats := &stats.MockHandlerStats{}
+	signatureRequestHandler := NewSignatureRequestHandler(warpBackend, message.Codec, mockHandlerStats)
+
+	tests := map[string]struct {
+		setup       func() (request message.SignatureRequest, expectedResponse []byte)
+		verifyStats func(t *testing.T, stats *stats.MockHandlerStats)
+	}{
+		"normal": {
+			setup: func() (request message.SignatureRequest, expectedResponse []byte) {
+				return message.SignatureRequest{
+					MessageID: messageID,
+				}, signature
+			},
+			verifyStats: func(t *testing.T, stats *stats.MockHandlerStats) {
+				assert.EqualValues(t, 1, mockHandlerStats.SignatureRequestCount)
+				assert.EqualValues(t, 1, mockHandlerStats.SignatureRequestHit)
+				assert.Greater(t, mockHandlerStats.SignatureRequestDuration, time.Duration(0))
+			},
+		},
+		"unknown": {
+			setup: func() (request message.SignatureRequest, expectedResponse []byte) {
+				return message.SignatureRequest{
+					MessageID: unknownMessageID,
+				}, nil
+			},
+			verifyStats: func(t *testing.T, stats *stats.MockHandlerStats) {
+				assert.EqualValues(t, 1, mockHandlerStats.SignatureRequestCount)
+				assert.EqualValues(t, 1, mockHandlerStats.SignatureRequestMiss)
+				assert.Greater(t, mockHandlerStats.SignatureRequestDuration, time.Duration(0))
+			},
+		},
+	}
+
+	for name, test := range tests {
+		// Reset stats before each test
+		mockHandlerStats.Reset()
+
+		t.Run(name, func(t *testing.T) {
+			request, expectedResponse := test.setup()
+			responseBytes, err := signatureRequestHandler.OnSignatureRequest(context.Background(), ids.GenerateTestNodeID(), 1, request)
+			assert.NoError(t, err)
+
+			// If the expected response is empty, assert that the handler returns an empty response and return early.
+			if len(expectedResponse) == 0 {
+				assert.Len(t, responseBytes, 0, "expected response to be empty")
+				return
+			}
+			var response message.SignatureResponse
+			if _, err = message.Codec.Unmarshal(responseBytes, &response); err != nil {
+				t.Fatal("error unmarshalling SignatureResponse", err)
+			}
```

Comment:
let's switch this test `require` (feel free to update the other files as well, for consistency)